ts/protocol/otf_message_handler.py

```diff
@@ -310,10 +312,27 @@ def _retrieve_input_data(conn):
 
     length = _retrieve_int(conn)
     value = _retrieve_buffer(conn, length)
-    if content_type == "application/json" and (decode_req is None or decode_req == "true"):
-        model_input["value"] = json.loads(value.decode("utf-8"))
-    elif content_type.startswith("text") and (decode_req is None or decode_req == "true"):
-        model_input["value"] = value.decode("utf-8")
+    if content_type == "application/json" and (
+        decode_req is None or decode_req == "true"
+    ):
+        try:
+            model_input["value"] = json.loads(value.decode("utf-8"))
+
+        except Exception as e:
+            model_input["value"] = value
+            logging.exception(e)
+            
+    elif content_type.startswith("text") and (
+        decode_req is None or decode_req == "true"
+    ):
+        try:
+            model_input["value"] = value.decode("utf-8")
+
+        except Exception as e:
+            model_input["value"] = value
+            logging.exception(e)
```

Comment:
```suggestion
            logging.warning("Failed utf-8 decoding of input data. Forwarding encoded payload", exc_info=True)
```

Reply:
Hmm but the issue here is you won't know what the payload was. The benefit of `logging.exception()` is that it'll print a stack trace

An alternative I was thinking of was

```python
logging.warning(f"Decoding failed with error {e} on payload {value})
```